Clarify and simplify Flow Get Started section, with a view to enhancing the onboarding experience for new users. (#950)
vaadin/docs: articles/flow/application/overview.asciidoc

Views can be nested.
For example, a main view could display sub-views.

<<events#, Handling Events>>::
Flow passes user interaction back to the server side as component events.
The application can handle these events in Java code using an event handler.
An event handler is an implementation of an event listener interface, such as a lambda expression or an anonymous class.
You can also use handler methods.

<<routing-navigation#, Basic Routing and Navigation>>::
Each view has a route through which it can be accessed.
A URL is used to select the required route.
Interaction with the application includes navigation between the different views.
+
For more information, see <<../routing/overview#, Routing and Navigation>>.

<<main-view#, Creating a Main View>>::
Applications often have a main view with a menu bar that shows a selection of sub-views.
When the user clicks on a menu item, the sub-view is displayed.
With the [classname]#AppLayout# component, you can create such views, which support navigation and responsive layouts.

<<resources#, Images and Icons>>::
Images and icons are basic visual features of an application.

> The Vaadin Design System comes with <<../../ds/foundation/icons#, two sets of icons>> that you can use in your application.

Reviewer: I’m not entirely sure about the rules when "the" should be used. In this case, "Vaadin Design System" could be considered as a product name, so it is capitalized. And in those cases my assumption has been that "the" is not used.

Author: I would say the reason that "the" is used is because the phrase "Vaadin Design System" contains the common noun "system". In the same way, we talk about THE Java Development Kit, even though it, too, is a product name and, hence, a proper noun (as a phrase).

<<styling#, Styling>>::

> To define the basic style of applications, Vaadin provides a default _Lumo_ theme, and an alternative _Material_ theme.

Reviewer: Did we have a guideline about using italics when emphasizing names? I’m curious about the reason why you chose to use it here. I do think it makes sense, but I’m not sure I would know when to use it myself.

Author: I made this change rather casually, it's true. There's no guideline on this currently. The wording "a default Lumo theme" sounds a little strange, as if some other kinds of Lumo theme existed; similarly with "an alternative Material theme". Would it be appropriate to say "... the Lumo theme by default, with the Material theme as an alternative."? We could remove the italics, since there's no basis in the guidelines for using it.

Context Dependency Injection (CDI) and Java EE::
The Java Enterprise Edition (EE) includes many features for creating enterprise applications.
CDI, or _context dependency injection_, is the Java EE way to manage service objects and inject them into applications, such as Vaadin UIs.
CDI requires a Java EE-enabled web container; the starter projects use Apache TomEE.
+
The starter project includes an example service that handles business data and logic.

> It is injected into the main view, and can be injected in a similar way into other views or elsewhere.

Reviewer: ```suggestion
The service injected into the main view, and can be injected in a similar way into other views or elsewhere.
```
